deps/apiinfo.go

```diff
@@ -155,3 +255,32 @@ func FullNodeProxy[T api.Chain](ins []T, outstr *api.ChainStruct) {
 		}
 	}
 }
+
+func Retry[T any](ctx context.Context, attempts int, initialBackoff time.Duration, errorTypes []error, f func(isRetry bool) (T, error)) (result T, err error) {
+	for i := 0; i < attempts; i++ {
+		if i > 0 {
+			log.Info("Retrying after error:", err)
+			time.Sleep(initialBackoff)
+			initialBackoff *= 2
+		}
+		result, err = f(i > 0)
+		if err == nil || !ErrorIsIn(err, errorTypes) {
+			return result, err
+		}
+		if ctx.Err() != nil {
+			return result, ctx.Err()
+		}
+	}
+	log.Errorf("Failed after %d attempts, last error: %s", attempts, err)
+	return result, err
+}
+
+func ErrorIsIn(err error, errorTypes []error) bool {
+	for _, etype := range errorTypes {
+		tmp := reflect.New(reflect.PointerTo(reflect.ValueOf(etype).Elem().Type())).Interface()
+		if errors.As(err, &tmp) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I'm not 100% convinced this actually works correctly, we definitely should test this PR somehow to make sure this / the rest of the stuff here actually works.

Can be as simple as starting with 2 lotus nodes and killing one while some RPC is ongoing (tricky, but possible with some e.g. message send where gas estimation takes a while)

Reply:
I am just a bit curious, why was the Lotus cluster functionality deprecated? Even if using Curio to replace lotus-miner, the Lotus cluster still seems useful.

Reply:
It pulled in massive unaudited raft dependencies, wasn't reliable, made signing endpoints available over libp2p with just simple allowlist filter to peers in the cluster, which was later discovered to not provide enough security because raft was never meant to run in an unsecured network context.

Basically it was a giant unmaintained hack which later on started to also produce a long list of security issues so it just didn't seem like a good idea to keep it around.